include/crm/common/logging.h

```diff
@@ -144,19 +168,19 @@ void pcmk_log_xml_impl(uint8_t level, const char *text, const xmlNode *xml);
  * \param[in] level  Priority at which to log the message
  * \param[in] fmt    printf-style format string literal for message
  * \param[in] args   Any arguments needed by format string
- *
- * \note This is a macro, and \p level may be evaluated more than once.
  */
 #  define do_crm_log(level, fmt, args...) do {                              \
-        switch (level) {                                                    \
+        uint8_t _level = pcmk__clip_log_level(level);                       \
```

Comment:
I know I mentioned using a leading underbar because it's pretty common in this situation but I forgot to mention it's also frowned upon. :) Technically the C standard reserves identifiers with a leading underbar in global scope (including macros). 

FYI, POSIX technically reserves all typedefs ending in `_t`, and I hope I don't see the day when the compiler enforces that. Even more ridiculous, as of C11, any function name like `str[a-z].*`, `is[a-z].*`, or `to[a-z].*` is reserved, among others.

https://www.gnu.org/software/libc/manual/html_node/Reserved-Names.html

Reply:
It seems about 30 years too late for the compiler to start enforcing those kinds of naming restrictions.

Reply:
I'm not 100% sure whether this applies here based on the standard. For starters, GNU's more readable explanation:
> "reserved names include all external identifiers (global functions and variables) that begin with an underscore (‘_’) "

doesn't quite match the explanation in the C17 draft standard PDF I'm looking at:
> "All identifiers that begin with an underscore are always reserved for use as identifiers with file scope in both the ordinary and tag name spaces"

(My emphasis is on "global" vs. "file scope".)

In either case, our variable here (`_level`) is in block scope within the macro, not global or file scope. If some variable `_level` is defined in file scope, this would override it only within the block, and there's no chance that anything within the block depends on the hypothetical file-scope `_level`.

If we had a macro called `_do_crm_log()` (such that the global macro name began with an underscore), then that would be a clear misuse of a reserved identifier. It might also be dangerous for one of the macro arguments to begin with an underscore (for example, `do_crm_log(_level, fmt, args...)`), given some of the weird ways that can behave.

---

The safer approach with regard to the standard is to avoid the use of a leading underscore in our variables altogether. It does make the macro definition somewhat more cumbersome... it means using some long/ugly variable name and/or telling callers via the doxygen not to pass in an argument via some particular variable name.

Why did they make these things so difficult...

Reply:
Ah you're right, this isn't covered since it's not an "external identifier." I just personally go by "never start with an underbar" to avoid having to remember the arcane rules. :) I'll merge this as-is

Reply:
One thought... if we ever want to be really paranoid about these things (including here -- I can open a follow-up PR), we could adopt a convention like `pcmk___` (triple underscore) to be used only for macro-scope variables. That way, by convention, a caller won't pass in a variable by that name, and we don't run the risk of using any reserved identifiers. Since the wording of the standard and the GNU doc aren't 100% clear.

Reply:
It's sad that that makes sense :)